subscription_oca/models/sale_subscription.py

```diff
@@ -0,0 +1,834 @@
+# Copyright 2023 Domatix - Carlos Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import datetime as dt
+import logging
+from datetime import date, datetime
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import _, api, fields, models
+from odoo.exceptions import AccessError
+from odoo.tools.misc import get_lang
+
+logger = logging.getLogger(__name__)
+ODOO_DATE_FORMAT = "%Y-%m-%d"
+
+
+class SaleSubscription(models.Model):
+    _name = "sale.subscription"
+    _description = "Subscription"
+    _inherit = ["mail.thread", "mail.activity.mixin"]
+    _order = "id desc"
+
+    color = fields.Integer("Color Index")
+
+    @staticmethod
+    def string2datetime(string_date):
+        return datetime.strptime(string_date, ODOO_DATE_FORMAT)
```

Comment:
For this operations you can use builtin to Date field methods
https://www.odoo.com/documentation/16.0/developer/reference/backend/orm.html#date-time-fields